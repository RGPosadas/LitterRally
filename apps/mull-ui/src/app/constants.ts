export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
