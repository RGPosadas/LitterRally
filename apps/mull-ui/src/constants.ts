import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
