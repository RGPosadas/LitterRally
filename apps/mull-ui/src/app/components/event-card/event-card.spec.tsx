import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { dummyEvent } from '../../../constants';
import EventCard from './event-card';

describe('EventCard', () => {
  const history = createMemoryHistory();
  it('should render successfully', () => {
    const { baseElement } = render(
      <MockedProvider>
        <Router history={history}>
          <EventCard event={dummyEvent} />
        </Router>
      </MockedProvider>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <MockedProvider>
          <Router history={history}>
            <EventCard event={dummyEvent} />
          </Router>
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render dates correctly', () => {
    const dom = render(
      <MockedProvider>
        <Router history={history}>
          <EventCard event={dummyEvent} />
        </Router>
      </MockedProvider>
    );

    const dateDiv = dom.getByTestId('event-card-datetime');
    expect(dateDiv.textContent).toBe('15 JUN01:45PM');
  });
});
