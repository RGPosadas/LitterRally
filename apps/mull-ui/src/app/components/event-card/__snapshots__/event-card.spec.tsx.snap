// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match snapshot 1`] = `
<div
  className="event-card-container button"
  style={Object {}}
>
  <img
    alt="Event"
    className="event-card-image"
    src="https://www.citywindsor.ca/residents/parksandforestry/City-Parks/PublishingImages/Assumption%20Park%20Street%20View.JPG"
  />
  <div
    className="event-card-datetime"
    data-testid="event-card-datetime"
  >
    <div
      style={
        Object {
          "fontWeight": "bolder",
        }
      }
    >
      1
       
    </div>
    <div>
      Jan
    </div>
    <div>
      12:00
    </div>
    <div>
      AM
    </div>
  </div>
  <button
    className="event-card-join "
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sign-in-alt fa-w-16 "
      data-icon="sign-in-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="event-card-description"
  >
    <div
      className="event-card-text"
    >
      <div
        className="event-card-title"
      >
        Test title
      </div>
      <div
        className="event-card-location"
      >
        15
        km â€¢ 
        1260 Remembrance Road Montreal, Qc
      </div>
    </div>
    <div
      className="event-members-div"
    >
      <img
        alt="participant_1"
        className="profile-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <img
        alt="participant_2"
        className="profile-picture"
        src="https://i.pinimg.com/236x/92/f0/ed/92f0edd9b0ecefdd5b7a48b8e1f7d340.jpg"
      />
      <img
        alt="participant_3"
        className="profile-picture"
        src="https://pbs.twimg.com/profile_images/748593045566853124/9DDVz0uT_400x400.jpg"
      />
    </div>
    <button
      className="event-card-share"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-share-alt fa-w-14 "
        data-icon="share-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;
